app/src/main/java/org/jellyfin/androidtv/preference/UserPreferences.kt

```diff
@@ -192,6 +192,11 @@ class UserPreferences(context: Context) : SharedPreferenceStore(
 		var defaultSubtitlesSize = Preference.int("subtitles_size", 28)
 	}
 
+	override operator fun <T : Preference<V>, V : Enum<V>> get(preference: T): V {
+		// Mock-able point for unit tests
+		return super.get(preference)
+	}
```

Comment:
Preference implementations should not override their parent functions, I don't even understand this one as it doesn't even do anything.

Reply:
So I'm a bit stumped how to resolve this
Removing this line causes the following exception:

```
java.lang.NullPointerException
	at org.jellyfin.androidtv.preference.SharedPreferenceStore.get(SharedPreferenceStore.kt:65)
	at org.jellyfin.androidtv.ui.playback.PlaybackControllerTest$prepDiMocks$1.invoke(PlaybackControllerTest.kt:38)
```

The cause of this is:
- We mock out `UserPreferences` to avoid `init` trying to run migrations on the shared preferences
- This implicitly mocks the parent abstract class `SharedPreferenceStore` so the internal val `sharedPreferences` is null (as the class constructor does not fire)
- As the return types are generic Mockk invokes the method with the passed types to determine the underlying types
- This naturally causes the real impl to run which throws a NPE

What I've tried:
- Mocking `SharedPreferences` and create a concrete `UserPreferences`, this triggers the migrations to run from init causing more problems
- Dropping type hints, the compiler cannot deduce the mocked types though
- Moving the type hints from `any` to `get<T, V>`, same problem
- Searching for the majority of the day if there's a way to workaround this from Mockk. But it appears there isn't.

So off the top of my head the solutions I've come up with (none of which I really like):
- Mark `UserPreferences` as open, create a shim with a setter to set the protected val (which needs changing to a var)
- Create a shim object which is passed to Koin, this will hold a real or injected `UserPreferencesStore` and forward to it
- Explicitly create an override with the types in, so Mockk does not have to deduce the types used (this current approach)

So I'm open to ideas or suggestions on this?

Reply:
After a couple of days of thinking, I realised we should just inject the interface. This matches traditional DI and resolves all the issues quite nicely without large sweeping changes